pkg/resources/instance/resource.go

```diff
@@ -349,6 +364,40 @@ func rCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag
 		}
 	}
 
+	// Attach blocks storage volumes if set
+	if bsSet, ok := d.Get(AttrBlockStorageVolumeIDs).(*schema.Set); ok {
```

Comment:
I feel like this section could be in its own function

Reply:
It makes sense, but i don't think it's worth the effort to refactor this resource as we are going to migrate the whole it to framework soon(ish).